### PR TITLE
Add animation to Submenu Tray when navigating between levels

### PR DESCRIPTION
**Pull request by reidbarber**
Adds a slide in/out animation when navigating between submenu level in the mobile tray experience.

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/WAI/ARIA/apg/)

## 📝 Test Instructions:

Test RSP Submenu and Menu Item Unavailable stories on mobile.

## 🧢 Your Project:

RSP

### CODE REVIEW

File: packages/@react-spectrum/menu/src/Menu.tsx [hunk 1]
```diff
@@ -70,13 +70,27 @@ function Menu<T extends object>(props: SpectrumMenuProps<T>, ref: DOMRef<HTMLDiv
...
   });
 
-  // TODO: add slide transition
+  let [traySubmenuAnimation, setTraySubmenuAnimation] = useState('');
+  let isMobile = useIsMobileDevice();
+  useEffect(() => {
+    if (!hasOpenSubmenu) {
+      setTraySubmenuAnimation('spectrum-TraySubmenu-enter');
+    }
+  }, [hasOpenSubmenu, isSubmenu]);
```
Comment (ktabors) [inline packages/@react-spectrum/menu/src/Menu.tsx:79]: Why is `isSubmenu` needed? Why is `isSubmenu` a variable that isn't state too?

**Review by ktabors — COMMENTED** [review]
Now that we have this nice animation should menu item unavailable also use it?

**Review by snowystinger — COMMENTED** [review]
When going back up a level, the old menu "snaps" in at the last second. Any way we can fade it in a little? or make it part of the animation and have it slide in from the left? (it could slide out as well when going down a level)

Otherwise, it is a big improvement!

**Comment by reidbarber** [comment]
@snowystinger just added a transition to the opacity, so that should be improved.

**Review by snowystinger — DISMISSED** [review]
lgtm, tested on iOS

File: packages/@react-spectrum/menu/src/ContextualHelpTrigger.tsx [hunk 2]
```diff
@@ -91,10 +98,13 @@ function ContextualHelpTrigger(props: InternalMenuDialogTriggerProps): ReactElem
...
-    if (parentMenuRef.current && !parentMenuRef.current.contains(document.activeElement)) {
-      parentMenuRef.current.focus();
-    }
+    setTraySubmenuAnimation('spectrum-TraySubmenu-exit');
+    setTimeout(() => {
+      submenuTriggerState.close();
+      if (parentMenuRef.current && !parentMenuRef.current.contains(document.activeElement)) {
+        parentMenuRef.current.focus();
+      }
+    }, 300); // Matches transition duration
```
Comment (LFDanLu) [inline packages/@react-spectrum/menu/src/ContextualHelpTrigger.tsx:107]: should this be 220 as well?

**Review by snowystinger — APPROVED** [review]

File: packages/@react-spectrum/menu/src/Menu.tsx [hunk 3]
```diff
@@ -136,6 +135,20 @@ export function TrayHeaderWrapper(props) {
...
+  let [traySubmenuAnimation, setTraySubmenuAnimation] = useState('');
+  useLayoutEffect(() => {
+    if (!hasOpenSubmenu) {
+      setTraySubmenuAnimation('spectrum-TraySubmenu-enter');
+    }
+  }, [hasOpenSubmenu, isMobile]);
+
+  let handleBackButtonPress = () => {
+    setTraySubmenuAnimation('spectrum-TraySubmenu-exit');
+    setTimeout(() => {
```
Comment (snowystinger) [inline packages/@react-spectrum/menu/src/Menu.tsx:147]: need to clean this up in the return of a useEffect

**Review by snowystinger — CHANGES_REQUESTED** [review]

**Review by snowystinger — APPROVED** [review]

**Review by LFDanLu — APPROVED** [review]

**Pull request merged**